xwayland/selection/selection.c

```diff
@@ -32,16 +45,59 @@ void xwm_selection_transfer_destroy_property_reply(
 	transfer->property_reply = NULL;
 }
 
-void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer) {
+void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer,
+		struct wlr_xwm_selection *selection) {
+	transfer->selection = selection;
 	transfer->wl_client_fd = -1;
 }
 
-void xwm_selection_transfer_finish(
+struct wlr_xwm_selection_transfer *xwm_selection_transfer_create_incoming(
```

Comment:
Maybe we should consider moving this to `incoming.c`.

Reply:
Done.